ssl/statem/statem_clnt.c

```diff
@@ -1688,7 +1689,8 @@ MSG_PROCESS_RETURN tls_process_server_certificate(SSL *s, PACKET *pkt)
     if ((SSL_IS_TLS13(s) && !PACKET_get_1(pkt, &context))
             || context != 0
             || !PACKET_get_net_3(pkt, &cert_list_len)
-            || PACKET_remaining(pkt) != cert_list_len) {
+            || PACKET_remaining(pkt) != cert_list_len
+            || PACKET_remaining(pkt) == 0) {
```

Comment:
Ah, I was misremembering what the empty Certificate the client sends when it declines to authenticate looks like (as this sentence is reiterating that the server cannot do so) -- it's not a select() block, so what you're checking for is indeed as empty as it gets.  Sorry for the noise.